### ISSUE TITLE
[2022 US Symposium] Allergies to Substances

### ISSUE THREAD

**Issue by ericaVoss** [OHDSI/Themis#69]
TYPE | NOTES
-- | --
ITEM | How to handle allergies to substances with link to the reaction.
FORUM POST | https://forums.ohdsi.org/t/2022-us-symposium-69-allergies-to-substances/17504
SOLUTION | TBD
NEXT STEPS | TBD

**Comment by cukarthik** [comment]
**Describe the issue/topic?**

     This proposal defines how to model allergies to substances within OMOP.

**What do we know about this topic? What has been discussed?**

      There are two cases when looking at allergies: (1) recording historical allergies and (2) allergic reactions that are newly experienced (i.e. recording rash after a vaccine is administered to a patient). The challenge on how to link the various components to (2) was primarily discussed. Looking at reactions to drugs, these tables have to be linked: condition_occurrence, observation, and drug_exposure.  To narrow the scope we will look at  drug allergies.

**What are recommendations for how to handle this issue/topic?**

     We agree with the proposed solution to historical allergies which was documented in the forum post: Drugs/allergens can be stored to OBSERVATION table to the column value_as_concept_id with one of the following observation_concept_id's = 439224 (Drug allergy) or 4169307 (Allergy to substance)

     The proposed solution is for CDM 5.4. For recording reactions to a drug, the following codes should be used for the observation_concept_id in the observation table: 4166257 (Allergic reaction to drug) or 441488 (Adverse reaction to substance). The value_concept_id will be the drug_concept_id from the drug_exposure table. The allergic reaction will be recorded in the condition_occurrence table (i.e. rash to drug). Right now to link all three tables is a challenge without the fact_relationship table. In 5.4, only one table can be linked to the observation table. Below is one option to link the information where the timestamp is used as an implicit linking of the condition_occurrence table and observation table, which is not a good solution. We should use an explicit link between the rows.

observation_datetime= condition_start_datetime
observation_event_id=drug_occurrence_id
obs_event_field_concept_id= 1147339 (drug_exposure)

If your data does not have drug information then observation and condition_occurrence table can be linked:

observation_datetime= condition_start_datetime
observation_event_id=condition_occurrence_id
obs_event_field_concept_id= 1147333 (condition_occurrence)


**What next steps should be taken?**

     TBD

**Comment by cgreich** [comment]
Friends:

I would not worry about having a link between the cause (drug, or other exposure) and the allergic response. Here is the why:

When we talk about "allergy", this layman term usually means a type I hypersensitivity reaction, also known as anaphylactic, allergic or immediate reaction. In these, it appears that the cause is relatively easy to pinpoint (patient took drug and the throat started swelling minutes later). Therefore, the model should absolutely be able to link between the event and the outcome.

But that picture is not true:

- There are three other types of hypersensitivity, and drugs can trigger those as well. They are usually slower (hours to days), and most folks would never even think of connecting the dots.
- Even for the type I hypersensitivity we cannot really know the cause. And no, the physician doesn't know for sure, either. Diagnosing them is more often than not inconclusive, because the tests (e.g. skin prick tests) love being false positive or false negative. 

But we need not worry. the OMOP CDM should not be used for treating individual patients anyway. We are looking at population effects, like when studying any other drug effect (think LEGEND). All we need is to have the exposure (we do) and the outcome (we do). One of the HADES methods will do the rest and determine the risk between them.

In other words: Let's not torture the poor ETL wretch into having to create an awkward assertion in the data. That's our job doing the research.

**Comment by MelaniePhilofsky** [comment]
@cukarthik @cgreich Looks like Karthik posts a solution, but then Christian posts an alteration to the solution. I think this should go back out to the forums for a final ok by the community before it is ratified. Here is the [link](https://ohdsiorg.sharepoint.com/:w:/s/Workgroup-CommonDataModel/EW1THso1H2VDodTymoY-eD8BE1Z46nNVUFpYjLvO_mdHJg?e=o1batd) to the forum post template. @cukarthik You are the sponsor for this issue, correct?

**Comment by cgreich** [comment]
the upcoming SNOMED Expo will have a session on this subject. Let's see what they came up with.

**Comment by Alexdavv** [comment]
There are a couple of points we might want to address here:
- There is a Drug Domain, but there are SNOMED [Substances](https://athena.ohdsi.org/search-terms/terms?conceptClass=Substance&standardConcept=Standard&vocabulary=OMOP+Extension&vocabulary=SNOMED&page=1&pageSize=500&query=) that are not drugs but could cause a reaction. Do we take them in? @cukarthik says we narrow the scope, but is that statement about it? In my opinion, these should be the same conventions with some minor changes.
- We should better define and distinguish the facts of allergic disposition (Observation) and current allergic reaction to the allergen (Condition). SNOMED concepts have this distinction.
- When we represent the fact of allergic disposition, do we distinguish "a finding of being allergic to a substance X" from "historical finding of unspecified allergic reactions to a substance X"? If no, we should only map to the allergic disposition-like concept(s), and explicitly prohibit mappings to historical concepts.
- Do we want to derive (by mapping to many) the allergic disposition information from the facts/historical facts of actual allergic reactions to a specific substance (e.g. "[Anaphylaxis caused by sulfur dioxide](https://athena.ohdsi.org/search-terms/terms/36713373)"? Or not, because they should always be the hierarchical descendants of "disposition" concepts? Do they? The same question goes to the allergic test results.
- Do we want at all the "unspecified allergic reactions to a substance X" or should we map them to "allergic disposition to a substance X'?
